src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogSourceCreationTests.cs

```diff
@@ -32,19 +30,136 @@ public void CheckSourceExistenceAndDeletion()
+        {
+            string source = new string('s', 254);
+            Assert.Throws<ArgumentException>(() => EventLog.CreateEventSource(source, null));
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void SourceDataNull()
+        {
+            Assert.Throws<ArgumentNullException>(() => EventLog.CreateEventSource(null));
+        }
 
-            Assert.Throws<System.ArgumentException>(() => EventLog.DeleteEventSource(Guid.NewGuid().ToString("N")));
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void SourceAlreadyExistsWhenCreatingSource()
+        {
+            string source = "Source_" + nameof(SourceAlreadyExistsWhenCreatingSource);
+            string log = "MyLog";
+            try
+            {
+                EventLog.CreateEventSource(source, log);
+                Assert.True(EventLog.SourceExists(source));
+                Assert.Throws<ArgumentException>(() => EventLog.CreateEventSource(source, log));
+            }
+            finally
+            {
+                EventLog.DeleteEventSource(source);
+                EventLog.Delete(log);
+            }
+        }
+
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void LogNameAlreadyExists_Throws()
+        {
+            string source = "Source_" + nameof(LogNameAlreadyExists_Throws);
+            string log = "AppEvent";
+
+            Assert.Throws<ArgumentException>(() => EventLog.CreateEventSource(source, log));
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void EventSourceCategoryCount_Invalid()
+        {
+            string log = "InvalidData";
+            string source = "Source_" + nameof(EventSourceCategoryCount_Invalid);
+
+            EventSourceCreationData mySourceData = new EventSourceCreationData(source, log);
+            Assert.Throws<ArgumentOutOfRangeException>(() => mySourceData.CategoryCount = -1);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void MessageResourceFile_Set()
+        {
+            string messageFile = "ResourceFile";
+            string source = "Source" + nameof(MessageResourceFile_Set);
+            string log = "MessageResourceFile";
+            EventSourceCreationData sourceData = new EventSourceCreationData(source, log);
+            sourceData.MessageResourceFile = messageFile;
+            Assert.Equal(messageFile, sourceData.MessageResourceFile);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void CategoryResourceFile_Set()
+        {
+            string messageFile = "ResourceFile";
+            string source = "Source" + nameof(MessageResourceFile_Set);
```

Comment:
these nameof's are all copy pasted and are the same. fix them to match their method names